tests/migrate_signals/test_operation_signals.py

```diff
@@ -0,0 +1,152 @@
+from unittest.mock import patch
+
+from django.db import connection
+from django.db.migrations import Migration
+from django.db.migrations.executor import MigrationExecutor
+from django.db.migrations.operations import RunPython
+from django.db.migrations.recorder import MigrationRecorder
+from django.db.models import signals
+from django.test import TransactionTestCase, override_settings
+
+from .operations import Operation0, Operation1, Operation2, Operation3
+
+
+class Receiver:
+    def __init__(self, operation):
+        self.sender = operation
+        signals.post_operation.connect(self, sender=operation)
+
+
+class InjectOperation1Receiver(Receiver):
+    def __call__(self, signal, sender, **kwargs):
+        return [Operation1()]
+
+
+class InjectOperation2Receiver(Receiver):
+    def __call__(self, signal, sender, **kwargs):
+        return [Operation2()]
+
+
+class InjectOperation1And2Receiver(Receiver):
+    def __call__(self, signal, sender, **kwargs):
+        return [Operation1(), Operation2()]
+
+
+class InjectOperation3Receiver(Receiver):
+    def __call__(self, signal, sender, **kwargs):
+        return [Operation3()]
+
+
+@override_settings(
+    MIGRATION_MODULES={"migrate_signals": "migrate_signals.noop_migrations"}
+)
+class PostOperationSignalTests(TransactionTestCase):
```

Comment:
You can probably subclass `TestCase` here instead and you should be able to remove the `# Reset applied-migrations state.` part in `tearDown`.

Reply:
Ah yes, I guess now that I'm not using the `migrate` management command I can simplify that, thanks.